File: apps/agor-android/app/src/main/kotlin/live/agor/app/models/Board.kt
package live.agor.app.models

import androidx.compose.runtime.Immutable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Serializable
data class Board(
    @SerialName("board_id") val boardId: String,
    val name: String,
    val description: String? = null,
    val emoji: String? = null,
    val color: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("created_by") val createdBy: String? = null,
    val archived: Boolean? = null,
)

File: apps/agor-rust-android/crates/agor/src/models/board.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Board {
    pub board_id: String,
    pub name: String,
    pub description: Option<String>,
    pub emoji: Option<String>,
    pub color: Option<String>,
    pub created_at: Option<String>,
    pub created_by: Option<String>,
    pub archived: Option<bool>,
}
